### PR TITLE
[ENG-2391] Add make-decision-dropdown component tests

### PR DESCRIPTION
**Pull request by fabmiz**
- Ticket: [ENG-2391]
- Feature flag: n/a

## Purpose

Add `make-decision-dropdown` component tests

## Summary of Changes
- Add tests
- Added `hasModeratorActions` to fix an edge-case where comment box and submit button show
despite no action trigger. The edge case occurs when `reviewsState` is `initial` and current user is a moderator and contributor on the registration.
For documentation purposes, I also added `withdrawn`, `rejected` even though this dropdown is currently not shown
in those `reviewsState`s

## QA Notes

[ENG-2391]: https://openscience.atlassian.net/browse/ENG-2391

### CODE REVIEW

File: lib/registries/addon/components/make-decision-dropdown/template.hbs [hunk 1]
```diff
@@ -14,50 +14,60 @@
...
+                            @id={{uniqueId}}
+                            @value={{option}}
+                            checked={{if (eq this.decisionTrigger option) true false}}
+                            {{on 'click' (fn this.updateDecisionTrigger option)}}
+                        />
+                        <label
+                            data-test-moderation-action-text={{option}}
+                            for={{uniqueId}}
+                        >
+                            {{humanize option}}
```
Comment (fabmiz) [inline lib/registries/addon/components/make-decision-dropdown/template.hbs:34]: use translated strings mapped to `ReviewActionTrigger`

File: translations/en-us.yml [hunk 2]
```diff
@@ -1362,6 +1362,7 @@ registries:
         additionalCommentPlaceholder: 'Add remarks to registration admins'
         justificationForWithdrawal: 'Justification for Withdrawal'
         justificationForWithdrawalPlaceholder: 'Provide justification for withdrawal'
+        noModeratorActions: 'No moderator decision can be made at this time'
```
Comment (futa-ikeda) [inline translations/en-us.yml:1365]:
```suggestion
        noModeratorActions: 'No moderator decisions can be made at this time'
```

**Review by futa-ikeda — COMMENTED** [review]
Super minor, and maybe a question for product?

File: lib/osf-components/addon/components/registries/review-actions-list/component.ts [hunk 3]
```diff
@@ -28,14 +29,13 @@ export default class ReviewActionsList extends Component<Args> {
...
         const { reviewActions } = this;
-        return (reviewActions || [])[0];
+        return A(reviewActions || []).objectAt(0);
     }
 
     @task({ withTestWaiter: true })
     fetchActions = task(function *(this: ReviewActionsList) {
         try {
-            const reviewActions = yield this.args.registration.reviewActions;
-            this.reviewActions = reviewActions.toArray();
```
Comment (fabmiz) [inline lib/osf-components/addon/components/registries/review-actions-list/component.ts:38]: turns out `toArray` does not guarantee order. See https://api.emberjs.com/ember/2.14/classes/Ember.Array/methods/toArray?anchor=toArray

File: mirage/config.ts [hunk 4]
```diff
@@ -242,10 +242,6 @@ export default function(this: Server) {
         relatedModelName: 'registration',
     });
     this.get('/providers/registrations/:parentID/registrations/', getProviderRegistrations);
-    osfNestedResource(this, 'registration-provider', 'actions', {
```
Comment (futa-ikeda) [inline mirage/config.ts:245]: any reason we are removing this? We don't use it now, but this is still a valid endpoint in the API.

Comment (fabmiz) [reply mirage/config.ts:245]: I think having unused code adds clutter and potential confusion. This is quite easy to add when it's needed.

Comment (futa-ikeda) [reply mirage/config.ts:245]: Truth. Then out it goes!

**Review by futa-ikeda — APPROVED** [review]
